nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -436,18 +451,15 @@ void RegulatedPurePursuitController::applyConstraints(
   if (use_cost_regulated_linear_velocity_scaling_ &&
     pose_cost != static_cast<double>(NO_INFORMATION))
   {
-    // Note(stevemacenski): We can use the cost at a pose as a derived value proportional to
-    // distance to obstacle since we lack a distance map. [0-128] is the freespace costed
-    // range, above 128 is possibly inscribed, so it should max out at 128 to be a minimum speed
-    // in when in questionable states. This creates a linear mapping of
-    // cost [0, 128] to speed [min regulated speed, linear vel]
-    double max_non_collision = 128.0;
-    if (pose_cost > max_non_collision) {
-      cost_vel = regulated_linear_scaling_min_speed_;
-    } else {
-      const double slope = (regulated_linear_scaling_min_speed_ - cost_vel) / max_non_collision;
-      cost_vel = slope * pose_cost + cost_vel;
-    }
+    auto inscribed_radius = costmap_ros_->getLayeredCostmap()->getInscribedRadius();
+
+    if (pose_cost != FREE_SPACE) {
+        const double lethal_dist = (-1.0 / inflation_cost_scaling_factor_) 
+        * std::log(pose_cost / (INSCRIBED_INFLATED_OBSTACLE - 1)) + inscribed_radius;
+        if (lethal_dist < cost_scaling_dist_){
```

Comment:
```suggestion
        if (lethal_dist < cost_scaling_dist_) {
```

just a linting thing